geometry/create-and-edit-geometries/src/main/java/com/esri/samples/create_and_edit_geometries/CreateAndEditGeometriesController.java

```diff
@@ -0,0 +1,591 @@
+    var geometry = geometryEditor.stop();
+    var graphic = new Graphic(geometry);
+
+    // set graphic style based on geometry type
+    switch (geometry.getGeometryType()) {
+      case POINT:
+        graphic.setSymbol(pointSymbol);
+        break;
+      case MULTIPOINT:
+        graphic.setSymbol(multipointSymbol);
+        break;
+      case POLYLINE:
+        graphic.setSymbol(lineSymbol);
+        break;
+      case POLYGON:
+        graphic.setSymbol(fillSymbol);
+        break;
+    }
+
+    // add new graphic to the graphics overlay
+    graphicsOverlay.getGraphics().add(graphic);
+  }
+
+  /**
+   * Updates a graphic already in the graphics overlay.
+   */
+  private void updateSelectedGraphic() {
+    // update the geometry of the graphic being edited and make it visible again
+    selectedGraphic.setGeometry(geometryEditor.stop());
+    selectedGraphic.setVisible(true);
+    selectedGraphic = null;
+  }
+
+  /**
+   * Stops the geometry editor without saving the geometry stored within.
+   */
+  @FXML
+  private void handleStopAndDiscardButtonClicked() {
+    // if editing a pre-existing graphic, make it visible again
+    if (selectedGraphic != null) {
+      selectedGraphic.setVisible(true);
+      selectedGraphic = null;
+      toolComboBox.setDisable(false);
+    }
+    geometryEditor.stop();
+  }
+
+  /**
+   * Removes all geometries from the graphics overlay.
+   */
+  @FXML
+  private void handleDeleteAllGeometriesButtonClicked() {
+    graphicsOverlay.getGraphics().clear();
+  }
+
+  /**
+   * Allows the user to select a graphic from the graphics overlay and edit its geometry.
+   */
+  private void selectGraphic() {
+
+    mapView.setOnMouseClicked(e -> {
+      graphicsOverlay.clearSelection();
+      Point2D mapViewPoint = new Point2D(e.getX(), e.getY());
+
+      // get graphics near the clicked location
+      ListenableFuture<IdentifyGraphicsOverlayResult> identifyGraphics;
+      identifyGraphics = mapView.identifyGraphicsOverlayAsync(graphicsOverlay, mapViewPoint, 10, false);
+
+      identifyGraphics.addDoneListener(() -> {
+        try {
+          // only select graphic if user not already creating/editing a geometry
+          if (!geometryEditor.isStarted()) {
+            if (!identifyGraphics.get().getGraphics().isEmpty()) {
+
+              // store the selected graphic
+              selectedGraphic = identifyGraphics.get().getGraphics().get(0);
+              selectedGraphic.setSelected(true);
+
+              // Use vertex tool if selected geometry is a point or multipoint
+              if (selectedGraphic.getGeometry().getGeometryType().equals(GeometryType.POINT) ||
+                selectedGraphic.getGeometry().getGeometryType().equals(GeometryType.MULTIPOINT)) {
+                geometryEditor.setTool(vertexTool);
+                toolComboBox.getSelectionModel().select(0);
+                toolComboBox.setDisable(true);
+              }
```

Comment:
```suggestion
```
Thanks to our bindings and setting the tool when we select point/multipoint, we can remove this block now.